server/data/historicalPrisonerApiClient.ts

```diff
@@ -9,22 +10,74 @@ export default class HistoricalPrisonerApiClient {
     return new RestClient('Historical Prisoner Api Client', config.apis.historicalPrisonerApi, token)
   }
 
-  findPrisonersWithIdentifiers(
+  async findPrisonersByNameOrAge(
     token: string,
     {
-      prisonerNumber,
-      pnc,
+      firstName,
+      lastName,
+      dateOfBirth,
+      ageFrom,
+      ageTo,
+      hdc = false,
+      lifer = false,
+      gender,
+      page,
+    }: {
+      firstName?: string
+      lastName?: string
+      dateOfBirth?: string
+      ageFrom?: string
+      ageTo?: string
+      hdc?: boolean
+      lifer?: boolean
+      gender?: string
+      page?: number
+    },
+  ): Promise<PagedModelPrisoner> {
+    logger.debug(`getting prisoner search details by name/age for `, {
```

Comment:
Added logging for all fields (including those not hooked up) to avoid eslint complaining about unused vars